### PR TITLE
chore: Twitter search switch

### PR DESCRIPTION
**Pull request by odilitime**
# Risks

Low

# Background

## What does this PR do?

reads and respects TWITTER_SEARCH_ENABLE (from .env or character json)

## What kind of change is this?

Improvements (misc. changes to existing features)

## Why are we doing this? Any context or related work?
So users don't have to edit source to enable this feature

# Documentation changes needed?

My changes require a change to the project documentation.

### CODE REVIEW

File: agent/src/index.ts [hunk 1]
```diff
@@ -335,6 +335,7 @@ export async function initializeClients(
     }
 
     if (clientTypes.includes("twitter")) {
+        TwitterClientInterface.enableSearch = getSecret(character, "TWITTER_SEARCH_ENABLE");
```
Comment (monilpat) [inline agent/src/index.ts:338]: Can we add this to the .env.example? TWITTER_SEARCH_ENABLE

**Review by monilpat — APPROVED** [review]
LGTM outside of adding to .env.example thanks for doing this :)

**Comment by odilitime** [comment]
Screen Shot 2024-12-11 at 11 00 40 PM
screenshots for the record

File: agent/src/index.ts (hunk 1 above)
Comment (odilitime) [reply agent/src/index.ts]: Added

**Pull request merged**